core/arch/arm/mm/core_mmu_lpae.c

```diff
@@ -381,6 +381,14 @@ static struct tee_mmap_region *init_xlation_table(struct tee_mmap_region *mm,
 		}
 		/* else Next region only partially covers area, so need */
 
+		if (mm->map) {
+			if (mm->map->region_size == 0)
+				mm->map->region_size = level_size;
+			else if (mm->map->region_size > level_size) {
```

Comment:
Since we are doing the same in both `if`and `else if` (i.e, `mm->map->region_size = level_size;`), why not combine them into one `if` statement and just one assignment?